### PR TITLE
WIP: update to include new specs

### PR DESCRIPTION
**Pull request by smlambert**
- add new builds (win32 & largeHeap) to release tool
- replace growing and soon-to-be awkward case statement with regex capture

Signed-off-by: smlambert <slambert@gmail.com>

### CODE REVIEW

File: sbin/Release.sh [hunk 1]
```diff
@@ -17,30 +17,23 @@ TIMESTAMP="$(date +'%Y%m%d')"
...
-        ARCH=aarch64 ;;
-    esac ;;
-    *Mac*)
-    OS=Mac && ARCH=x64 && EXT=tar.gz ;;
-    *AIX*)
-    OS=AIX && ARCH=ppc64 && EXT=tar.gz ;;
-  esac
+  re="OpenJDK([a-zA-Z0-9]+)_([a-zA-Z0-9]+)_([a-zA-Z0-9]+)_([a-zA-Z0-9]+)";
+  EXT=tar.gz;
+  if [[ $f =~ $re ]];
```
Comment (karianna) [inline sbin/Release.sh]: suggest using full variable names, e.g. `EXTENSION`, `file` and maybe replacing `re` with the semantics of the pattern you're matching, maybe `fileprefix` or something?

Comment (karianna) [reply sbin/Release.sh]: Should this regex refactoring also be repeated below?

**Comment by gdams** [comment]
FYI a further change will be required in the website repo to add a new platform block https://github.com/AdoptOpenJDK/openjdk-website/blob/master/src/json/config.json#L49-L59

**Comment by smlambert** [comment]
Yes, this PR is marked as a WIP, I will be getting rid of the case statements below also.  Still actively working on it...

Require ci server back up so I can check the naming of all builds, and all build artifacts to ensure all t's are crossed, all i's dotted.  :)

File: sbin/Release.sh (hunk 1 above)
Comment (smlambert) [reply sbin/Release.sh]: re = regular expression, the entire one I am trying to match, and capture each segment into a variable.  Requires naming convention of builds to be adhered to... (though then again, so did the case statement approach)

Comment (smlambert) [reply sbin/Release.sh]: One other thing to note, this script generates a timestamp upon being run, and it gets used as the one that is published, while the timestamp that is created and added to the build artifact at the time of the build is ignored (is that intentional)?

File: sbin/Release.sh [hunk 2]
```diff
@@ -13,84 +13,40 @@
...
-TIMESTAMP="$(date +'%Y%m%d')"
 npm install
-for f in OpenJDK*.tar.gz
+# Expect a naming convention for build artifacts to follow this pattern:
+# OpenJDK<version>_<arch>_<os>_<timestampOrTag>.<extension>
+# Examples: OpenJDK8_x64_Windows_201813060547.zip, 
+#           OpenJDK8_x64_LinuxLH_201813060547.tar.gz, 
+#           OpenJDK10_aarch64_Linux_201813060547.sha256.txt
+#           OpenJDKamber_x64_Linux_201813061304.tar.gz	
+for f in OpenJDK*
```
Comment (karianna) [inline sbin/Release.sh]: Minor nitpick, what is a 'f'? :-)

File: sbin/Release.sh [hunk 3]
```diff
@@ -13,84 +13,40 @@
...
-    OS=Mac && ARCH=x64 && EXT=tar.gz ;;
-    *AIX*)
-    OS=AIX && ARCH=ppc64 && EXT=tar.gz ;;
-  esac
-  if [ "$REPO" == "releases" ]; then
-    mv $f "Open${VERSION}_${ARCH}_${OS}_${TAG}.${EXT}"
-    elif [ "$REPO" == "nightly" ]; then
-    mv $f "Open${VERSION}_${ARCH}_${OS}_$TIMESTAMP.${EXT}"
+#              1)VERSION 2)ARCH         3)OS           4)TS_TAG       5)EXTENSION 
+  regex="OpenJDK([a-zA-Z0-9]+)_([a-zA-Z0-9]+)_([a-zA-Z0-9]+)_([a-zA-Z0-9]+).(tar.gz|sha256.txt|zip)";
```
Comment (karianna) [inline sbin/Release.sh]: is the convention to use the local keyword in front of a local bash var?

File: sbin/Release.sh (hunk 2 above)
Comment (smlambert) [reply sbin/Release.sh]: f is the variable some other person used, but agree, will update to 'file'  ;)

File: sbin/Release.sh (hunk 3 above)
Comment (smlambert) [reply sbin/Release.sh]: I was not sure if we(Adopt) followed a programming convention for shell scripts, based on your question, I looked up a commonly followed one (https://google.github.io/styleguide/shell.xml), and note that they use local for variables in functions.  So will leave as is, but in general will try to comply with the google style guide for shell scripting.

**Comment by karianna** [comment]
@smlambert safe to merge or do you have more to go in?

File: sbin/Release.sh (hunk 3 above)
Comment (smlambert) [reply sbin/Release.sh]: thanks for the pointer to shellcheck too

**Comment by smlambert** [comment]
The changes in this PR imply that I will be updating the names of some build artifacts, so that follow the expected naming convention:
- artifacts that are temporarily named OpenJDK_amber_* would renamed openjdkamber_* so that amber is the < version > 
- artifacts that are temporarily named OpenJDK8_x64_Linux_largeHeap_* would be renamed OpenJDK8_x64_LinuxLH_*

When @johnoliver changes come in, he has added 'impl' to the name, in a standard way, so we can easily parse/grab that info by updating the regex in Release.sh

Note that near the end of the Release.sh script all temporary names appear to be standardized (see sed command in the script).  I left as is.

**Comment by smlambert** [comment]
I will merge this shortly, I need to update the Jenkins config of 2 jobs first (as they do not adhere to the naming convention outlined in Release.sh)

**Comment by smlambert** [comment]
Ok, I have updated the Jenkins configurations that I think I need to at this time.  I will watch over the next run of the nightly pipelines for any surprises or things I may have missed.

I'll also follow this with some updated doc around build artifacts, and naming conventions.

**Pull request merged**